src/utils.rs

```diff
@@ -162,6 +169,81 @@ impl Array {
     }
 }
 
+pub(crate) struct MlxVectorArray {
```

Comment:
One small comment -- we call MlxArray just Array, we can probably call this just VectorArray

Reply:
Sure. I chose `MlxVectorArray` just so that it kinda tells us it's just a wrapper over c, but I guess `Array` is similar. Addressed in f6632fdaacecd6d521d1308086647f51608abc9a